pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -352,3 +378,19 @@ func isMaster(machine mapiv1.Machine, client client.Client) bool {
 	}
 	return false
 }
+
+func getClusterInfrastructure(c client.Client) (*configv1.Infrastructure, error) {
+	infra := &configv1.Infrastructure{}
+	key := types.NamespacedName{
+		Namespace: metav1.NamespaceNone,
+		Name:      "cluster",
+	}
+	if err := c.Get(context.TODO(), key, infra); err != nil {
```

Comment:
I would prefer to use informer here so you don't have to query the apiserver everytime you request for infrastructure object. Plus, as Alberto says, you can check the infrastructure platform only once since it's unlikely we will support hybrid installations in 4.2. So calling it only once will do as well.

Reply:
I think if I will fetch it only once it will be good enough also without informer

Reply:
the controller runtime client already use an informer cache underneath, also see here for an example of how to use indexers https://github.com/openshift/machine-api-operator/blob/master/pkg/controller/nodelink/nodelink_controller.go#L141

Reply:
see https://github.com/openshift/machine-api-operator/pull/346/files#r302447055